Bottom: c5cff54b92786f7aee8d4dd6825c2b48b572396f
Top:    e70b38aa1fa5f9909e903d95d2db6d19a62127c1
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-12-17 21:38:36 +0000

drc: HACK HACK HACK


---

diff --git a/src/find.c b/src/find.c
index 1a3bb8f..4f28739 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2508,7 +2508,12 @@ IsPolygonInPolygon (PolygonType *P1, PolygonType *P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = Bloat;
+              /* Another Bloat is added by IsLineInPolygon, making the correct
+               * 2x Bloat. Ideally we would change it there, but doing so
+               * breaks some other DRC checks which rely on the broken behaviour
+               * in IsLineInPolygon.
+               */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
